bson/primitive/example_test.go

```diff
@@ -0,0 +1,50 @@
+package primitive_test
+
+import (
+	"context"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+func ExampleRegex() {
```

Comment:
Would this be more useful as an example on [Collection.Find](https://pkg.go.dev/go.mongodb.org/mongo-driver/mongo#Collection.Find) if users are looking there first for how to define filters? Or maybe a link to the `primitive` types that can be used in filters so they can discover this example?